api/clients/config_v2.go

```diff
@@ -0,0 +1,17 @@
+package clients
+
+import (
+	"github.com/Layr-Labs/eigenda/api/clients/codecs"
+)
+
+// EigenDAClientConfigV2 contains configuration values for EigenDAClientV2
+type EigenDAClientConfigV2 struct {
```

Comment:
Starting this thread to discuss package structure. I personally don't like that all our clients are in the same package. I think I would prefer if each client was in its own separate package, because that allows to have cleaner names, and make sure to prevent clashes. For eg:
```
import "...clients/v2/disperser"
import edaclient ".../v2/eigenda"
func main() {
   dispClient := disperser.New(disperser.Config{...})
   edaclient.New(edaclient.Config{...}, dispClient)
}
```
is way cleaner than having long names all under clients package.

I'm fine if people prefer to keep all the clients in same namespace, but at least I think we should separate the v1 and v2 namespace. Thoughts? @ian-shim @epociask @bxue-l2 @jianoaix

Reply:
Makes sense to me. We are getting a lot "v2" in the names around.